src/security/ldap/src/main/java/org/geoserver/security/ldap/LDAPBaseSecurityServiceConfig.java

```diff
@@ -50,6 +50,8 @@ public abstract class LDAPBaseSecurityServiceConfig extends BaseSecurityNamedSer
 
     String adminGroup;
     String groupAdminGroup;
+    String rolePrefix = "ROLE_";
```

Comment:
Is this class serialized/deserialized using XStream? If so, there might be issues with people upgrading from an older data directory, where the fields were missing.

XStream bypasses the normal object construction and these fields initializers won't be used, so defaults are not reliable. For the ``convertToUpperCase`` you can recognize ``null`` in the getter and make it "true" (to be done), but for the roleprefix it seems different, not sure there is a difference between the lack of XML tag, and possibly and empty tag, if someone configures it to have no prefix at all. Something you should check, I don't have a ready-made solution for it.

Reply:
Fixed so that nulls convert to defaults (and I validated that the lack of a tag in the XML is returned as null and the empty tag is an empty string) and empty string is returned by wicket when the field is empty.